### PR TITLE
Have a mocked up environment for developing locally

### PR DESCRIPTION
**Pull request by emettely**
**Is your Pull Request request related to another [issue](https://github.com/bbc/react-transcript-editor/issues) in this repository ?**      

Having a look over the Firebase docs, it looks like many more features are supported for emulators!
Which means a lot of the scripts and dependencies to run servers in parallel can now be simplified.

I have been working from an older version of docs when speaking about their testing strategy - they encourage to use emulators over `serve`.  See https://firebase.google.com/docs/hosting/test-preview-deploy and https://medium.com/firebase-developers/setting-up-a-fully-functional-database-in-firebase-emulator-b0199fff0252

**Describe what the PR does**    

- [x] `npm run start` should now start the emulator, use the emulated DB and have a hot-reloading local version.
- [x] removed unnecessary deps
- [x] I've updated the browser API key in GCP - to accept from `localhost:5000/*` as well, in case you want to run the build version using the emulator.
- [x] Import dummy data when running emulator
- [x] updated firebase dependencies
- [x] mechanism to update firestore security rules. Also see https://github.com/bbc/digital-paper-edit-firebase/issues/189#issuecomment-816702242 for context as to why it's copying over the security rules

**State whether the PR is ready for review or whether it needs extra work**    
Ready

**Additional context**

### CODE REVIEW

File: src/Components/Firebase/Collection.js [hunk 1]
```diff
@@ -29,6 +29,7 @@ class Collection {
 
   postItem = async data => {
     try {
+      console.log(this.collectionRef);
```
Comment (emettely) [inline src/Components/Firebase/Collection.js:32]:
```suggestion
```

File: src/Components/Firebase/firebase.js [hunk 2]
```diff
@@ -14,21 +14,36 @@ const config = {
...
-    const dbSnapshot = await dbUserRef.get();
-    const dbUser = dbSnapshot.data();
+    let dbSnapshot = { exists: false };
+    let dbUser = {};
+
+    try {
+      dbSnapshot = await dbUserRef.get();
+      dbUser = dbSnapshot.data();
+    } catch (err) {
+      console.log(err);
```
Comment (emettely) [inline src/Components/Firebase/firebase.js:45]:
```suggestion
      console.error(err);
```

File: src/Components/Projects/index.js [hunk 3]
```diff
@@ -33,11 +33,14 @@ const Projects = (props) => {
...
 
   const createProject = async (item) => {
+    console.log(item);
     const docRef = await projectsCollection.postItem(item);
     docRef.update({
       url: `/projects/${ docRef.id }`,
     });
 
+    console.log(docRef);
+
```
Comment (emettely) [inline src/Components/Projects/index.js:43]:
```suggestion
```

File: src/Components/Projects/index.js [hunk 4]
```diff
@@ -33,11 +33,14 @@ const Projects = (props) => {
   };
 
   const createProject = async (item) => {
+    console.log(item);
```
Comment (emettely) [inline src/Components/Projects/index.js:36]:
```suggestion
```

File: src/Components/Projects/index.js [hunk 5]
```diff
@@ -59,6 +62,7 @@ const Projects = (props) => {
     } else {
       item.users = [ uid ];
       item.url = '';
+      console.log(uid);
```
Comment (emettely) [inline src/Components/Projects/index.js:65]:
```suggestion
```

File: src/Components/Projects/index.js [hunk 6]
```diff
@@ -153,7 +157,7 @@ const Projects = (props) => {
             />
           </Col>
         </Row>
-        {items ? (
+        {items && uid ? (
```
Comment (emettely) [inline src/Components/Projects/index.js:160]:
```suggestion
        {items ? (
```

**Comment by allishultes** [comment]
When running this locally, it doesn't let me open a project to advance to the next workspace view... so the project hyperlinks aren't working
Screenshot 2021-04-12 at 10 14 23

**Comment by emettely** [comment]
> When running this locally, it doesn't let me open a project to advance to the next workspace view... so the project hyperlinks aren't working
> Screenshot 2021-04-12 at 10 14 23

You need to use your fake user - test@email.com password :)

**Comment by emettely** [comment]
@leihelondon as discussed this morning - we'd need a ticket for setting up a dummy storage and transcription as it's not implemented as part of the emulators yet...
Screenshot 2021-04-12 at 10 21 00

**Comment by leihelondon** [comment]
As discussed, moving this to backlog until we have a solution for emulating storage.

**Pull request closed without merging**